packages/hub/src/components/GlobalSearch/index.tsx

```diff
@@ -0,0 +1,175 @@
+"use client";
+import { clsx } from "clsx";
+import { useRouter } from "next/navigation";
+import { FC, useCallback, useEffect, useRef } from "react";
+import { fetchQuery, graphql, useRelayEnvironment } from "react-relay";
+import {
+  components,
+  DropdownIndicatorProps,
+  SelectInstance,
+} from "react-select";
+import AsyncSelect from "react-select/async";
+
+import { SearchIcon } from "@quri/ui";
+
+import { GlobalSearchQuery } from "@/__generated__/GlobalSearchQuery.graphql";
+import { SearchResult$key } from "@/__generated__/SearchResult.graphql";
+import { SearchResult } from "./SearchResult";
+import { SearchResultEdge$key } from "@/__generated__/SearchResultEdge.graphql";
+
+export const Query = graphql`
+  query GlobalSearchQuery($text: String!) {
+    result: search(text: $text) {
+      __typename
+      ... on BaseError {
+        message
+      }
+      ... on QuerySearchConnection {
+        edges {
+          cursor
+          ...SearchResultEdge
+          node {
+            id
+            link
+            object {
+              ...SearchResult
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export type SearchOption =
+  | {
+      type: "object";
+      id: string;
+      link: string;
+      edge: SearchResultEdge$key;
+      object: SearchResult$key;
+    }
+  | {
+      type: "error";
+      message: string;
+    };
```

Comment:
Kinda sad you didn't use a Result here

Reply:
- I forgot about it
- using a type from squiggle-lang seems wrong
- also it's possible we'll have more than two types, e.g. I can imagine local history autocomplete options being in the list

Reply:
I'd like us to get used to a small Standard Library for basic things. RIght now that's in Squiggle-lang, but later we could pull out.

Result is a really fundamental type.

That said, if we have more than two types later, that's a good reason not to use Result.